### PR TITLE
XdgDesktopFile: Adds a Desktop File ID calculator method

### PR DESCRIPTION
**Pull request by luis-pereira**
It implements the desktop-file ID calculator.
Spec:
https://specifications.freedesktop.org/desktop-entry-spec/latest/ape.html

### CODE REVIEW

**Comment by jleclanche** [comment]
gtm

File: xdgdesktopfile.cpp [hunk 1]
```diff
@@ -1072,6 +1073,37 @@ bool checkTryExec(const QString& progName)
...
+    const QStringList dataDirs = XdgDirs::dataDirs();
+
+    foreach(const QString &d, dataDirs) {
+        if (id.startsWith(d)) {
+            // remove only the first occurence
+            id.replace(id.indexOf(d), d.size(), QString());
+        }
+    }
+
+    const QLatin1Char dash('/');
```
Comment (paulolieuthier) [inline xdgdesktopfile.cpp]: You mean *slash*?

**Comment by paulolieuthier** [comment]
GTM.

File: xdgdesktopfile.cpp (hunk 1 above)
Comment (luis-pereira) [inline xdgdesktopfile.cpp]: yeap. thanks @paulolieuthier

**Pull request merged**